tests/listobjects/listobjects.go

```diff
@@ -115,23 +118,59 @@ func runTests(t *testing.T, schemaVersion string, client ListObjectsClientInterf
 					if assertion.ErrorCode == 0 {
 						require.NoError(t, err)
 						require.Subset(t, assertion.Expectation, resp.Objects)
+					} else {
+						require.Error(t, err)
+						e, ok := status.FromError(err)
+						require.True(t, ok)
+						require.Equal(t, assertion.ErrorCode, int(e.Code()))
+					}
+
+					// assert 2: on streaming list objects endpoint
+					done := make(chan struct{})
+					var streamedObjectIds []string
 
-						for _, object := range resp.Objects {
-							// each object in the response of ListObjects should return check -> true
-							checkResp, err := client.Check(ctx, &pb.CheckRequest{
-								StoreId:          storeID,
-								TupleKey:         tuple.NewTupleKey(object, assertion.Request.Relation, assertion.Request.User),
-								ContextualTuples: assertion.Request.ContextualTuples,
-							})
-							require.NoError(t, err)
-							require.True(t, checkResp.Allowed, fmt.Sprintf("Expected Check(%s#%s@%s) to be true, got false", object, assertion.Request.Relation, assertion.Request.User))
+					clientStream, err := client.StreamedListObjects(ctx, &pb.StreamedListObjectsRequest{
+						StoreId:          storeID,
+						Type:             assertion.Request.Type,
+						Relation:         assertion.Request.Relation,
+						User:             assertion.Request.User,
+						ContextualTuples: assertion.Request.ContextualTuples,
+					}, []grpc.CallOption{}...)
+					go func() {
+						for {
+							resp, err := clientStream.Recv()
+							if err != nil {
+								break
+							}
+							streamedObjectIds = append(streamedObjectIds, resp.Object)
 						}
+						done <- struct{}{}
```

Comment:
Will a more "go" approach be passing the err and streamedObjectIds to the channel?

Reply:
That would be unnecessary, `done <- struct{}{}` is used to signal that `streamedObjectIds` has received all the objects  and that we can proceed with the assertions after line 149

I will add code that checks that `err is EOF`. If it's not we have a problem in the Streaming code.